common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1927,9 +1930,27 @@ def _create_courselike(
 
         elif block_fields or definition_fields:  # pointing to existing course w/ some overrides
             # just get the draft_version structure
+
             draft_version = CourseLocator(version_guid=versions_dict[master_branch])
             draft_structure = self._lookup_course(draft_version).structure
             draft_structure = self.version_structure(locator, draft_structure, user_id)
+
+            # This is the point where we're editing the draft structure before creating a new course.
+            # So we'll want look for all dates (start, end, due, beta, etc.) and move them by the
+            # difference between the original start datetime and the new requested startdate time.
+            #
+            # A few notes:
+            #
+            #  * Whether we choose local or UTC time? One is more intuitive, but it would suck if
+            #    they did this only to have to manually change all the assignment times by a few
+            #    hours to align to what it was last run.
+            #  * ORA2 has an extra datetime differentiating the submission vs. assessment periods.
+            #    It sucks to special case this, but it might be necessary. I don't think the current
+            #    code really handles it at any rate.
+            #  * By the time the block field data gets passed in here (block_fields), it's already
+            #    got the new datetimes for start -- to get the original course start (to calculate
+            #    the offset), we might have to pass it in explicitly.
```

Comment:
By block_field data, I mean the field data for the root course. Once we have the offset, we'd loop over all the blocks in draft_structure['blocks'] to move the dates. This is at a layer lower than where inheritance gets applied, so we shouldn't have to worry about that aspect at least.